packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -24,6 +24,70 @@ export interface PreconditionChecker {
   check(): boolean;
 }
 
+export interface PostconditionChecker<T> {
+  check(
+    inputs: ContinueResponse<T> | CancelResponse
+  ): Promise<ContinueResponse<T> | CancelResponse>;
+}
+
+export class FilePathExistsChecker
+  implements PostconditionChecker<DirFileNameSelection> {
+  private fileExtension: string;
+
+  public constructor(fileExtension: string) {
+    this.fileExtension = fileExtension;
+  }
+  public getGlobPattern(
+    inputs: ContinueResponse<DirFileNameSelection>
+  ): string {
+    const lightningExtensions = ['.app', '.cmp', '.intf', '.evt'];
+    // Lightning creates a top level directory with the name of the
+    // file entered so the glob pattern needs to be different
+    if (lightningExtensions.indexOf(this.fileExtension) !== -1) {
+      return path.join(
+        inputs.data.outputdir,
+        inputs.data.fileName,
+        inputs.data.fileName + this.fileExtension
+      );
+    } else {
+      return path.join(
+        inputs.data.outputdir,
+        inputs.data.fileName + this.fileExtension
+      );
+    }
+  }
+  public async check(
+    inputs: ContinueResponse<DirFileNameSelection> | CancelResponse
+  ): Promise<ContinueResponse<DirFileNameSelection> | CancelResponse> {
+    if (inputs.type === 'CONTINUE') {
+      const globPattern = this.getGlobPattern(inputs);
+      const files = await vscode.workspace.findFiles(globPattern);
+      // If file does not exist then create it, otherwise prompt user to override the file
+      if (files.length === 0) {
+        return inputs;
+      } else {
+        const override = await notificationService.showWarningMessage(
+          nls.localize('warning_prompt_file_override'),
+          nls.localize('warning_prompt_yes'),
+          nls.localize('warning_prompt_no')
+        );
+        if (override === 'Yes') {
```

Comment:
I'm not sure if this will always be 'Yes'. 
If you call nls.localize on the message before showing it, won't it be whatever the localized message is?